iac-extensions/arm/src/test/java/org/sonar/iac/arm/tree/impl/json/ResourceDeclarationImplTest.java

```diff
@@ -256,4 +257,45 @@ void shouldParseResourceWithComplexProperties() {
     assertThat(property.key().value()).isEqualTo("sourceAddressPrefixes");
     assertThat(property.value()).hasKind(ARRAY_EXPRESSION).hasArrayExpressionValues("0.0.0.0/0");
   }
+
+  @Test
+  void shouldParseResourceWithChildResourcesInIt() {
+    String code = code("{",
+      "  \"resources\": [",
+      "    {",
+      "      \"name\": \"parent resource\",",
+      "      \"type\": \"Microsoft.Network/networkSecurityGroups\",",
+      "      \"apiVersion\": \"2022-11-01\",",
```

Comment:
Is the coma at the end of the string expected here?

Reply:
Yes, as we have the next attributes `resources` coming at the next line.

Reply:
```suggestion
      "      \"name\": \"parent resource\"",
      "      \"type\": \"Microsoft.Network/networkSecurityGroups\"",
      "      \"apiVersion\": \"2022-11-01\"",
```

Reply:
ok sorry, nevermind